backend/src/main/java/com/example/dontwastetomuch/game/GameService.java

```diff
@@ -75,7 +76,19 @@ public List<GameData> getAllMyGames(MyUser myUser) {
         return myUser.getGameData();
     }
 
-    public void updateGameStats(MyUser user) {
+    public void updateGameStats(MyUser user, NewStatsDTO newStatsDTO) {
+        double playtime = user.getGameData().stream().filter(game -> newStatsDTO.getGameId().equals(game.getGameId())).findAny().orElseThrow().getPlaytime();
+        double spentMoneyGame = user.getGameData().stream().filter(game -> newStatsDTO.getGameId().equals(game.getGameId())).findAny().orElseThrow().getSpentMoneyGame();
```

Comment:
You could store this part…
```
user.getGameData().stream().filter(game -> newStatsDTO.getGameId().equals(game.getGameId())).findAny().orElseThrow()
```
…in a seperate variable and replace it in all the upcoming calls.